src/core/simulation.jl

```diff
@@ -615,68 +601,68 @@ function build!(
 end
 
 function _build!(sim::Simulation, serialize::Bool)
-    sim.internal.build_status = BuildStatus.IN_PROGRESS
-    stage_initial_times = _get_simulation_initial_times!(sim)
+    set_simulation_build_status!(sim, BuildStatus.IN_PROGRESS)
+    problem_initial_times = _get_simulation_initial_times!(sim)
     sequence = get_sequence(sim)
-    for (stage_number, stage_name) in get_order(sequence)
-        stage = get_stage(sim, stage_name)
-        if stage === nothing
-            throw(
-                IS.ConflictingInputsError(
-                    "Stage $(stage_name) not found in the stages definitions",
-                ),
-            )
+    for (ix, (problem_name, problem)) in enumerate(get_problems(sim))
+        problem_interval = get_interval(sequence, problem_name)
+        # Note to devs: Here we are setting the number of operations problem executions we
+        # will see for every step of the simulation
+        if ix == 1
+            set_executions!(problem, 1)
+        else
+            step_resolution = get_step_resolution(sequence)
+            get_interval(sequence, problem_name)
+            set_executions!(problem, Int(step_resolution / problem_interval))
+            Int(step_resolution / problem_interval)
         end
-        stage_interval = get_stage_interval(sim, stage_name)
-        step_resolution =
-            stage_number == 1 ? get_step_resolution(sim.sequence) :
-            get_stage_interval(sequence, get_order(sequence)[stage_number - 1])
-        stage.internal.executions = Int(step_resolution / stage_interval)
-        stage.internal.number = stage_number
-        stage.internal.name = stage_name
-        _attach_feedforward!(sim, stage_name)
+        _attach_feedforward!(sim, problem_name)
     end
     _assign_feedforward_chronologies(sim)
-    _check_steps(sim, stage_initial_times)
-    _build_stages!(sim)
+    _check_steps(sim, problem_initial_times)
+    _build_problems!(sim)
     if serialize
         TimerOutputs.@timeit BUILD_SIMULATION_TIMER "Serializing Simulation Files" begin
-            serialize_simulation(sim)
+            # serialize_simulation(sim)
```

Comment:
Perhaps mark this somehow so that it doesn't get lost?

Reply:
Yeah, this needs to be reimplemented